src/JitInterface/src/CorInfoImpl.cs

```diff
@@ -3453,9 +3453,11 @@ private void recordRelocation(void* location, void* target, ushort fRelocType, u
             BlockType locationBlock = findKnownBlock(location, out relocOffset);
             Debug.Assert(locationBlock != BlockType.Unknown, "BlockType.Unknown not expected");
 
+#if !ARM
```

Comment:
We are trying to avoid ifdefs in the compiler. 

Also, I think it would be better to just return here instead of recording a bogus reloc. It will make the crash easier to debug.

It should look something like this:

```
if (locationBlock != BlockType.Code)
{
    // TODO: https://github.com/dotnet/corert/issues/3877
    TargetArchitecture targetArchitecture = _compilation.TypeSystemContext.Target.Architecture;
    if (targetArchitecture == TargetArchitecture.ARM || targetArchitecture == TargetArchitecture.ARMEL)
        return;
    throw new NotImplementedException("Arbitrary relocs"); 
}
```